ts/features/mvl/screens/details/components/attachment/MvlAttachmentDownload.tsx

```diff
@@ -0,0 +1,184 @@
+import FooterWithButtons from "../../../../../../components/ui/FooterWithButtons";
+import i18n from "../../../../../../i18n";
+import { useIODispatch, useIOSelector } from "../../../../../../store/hooks";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import { MvlAttachment } from "../../../../types/mvlData";
+import { mvlPreferencesSetWarningForAttachments } from "../../../../store/actions";
+import { showToast } from "../../../../../../utils/showToast";
+import { mvlAttachmentDownload } from "../../../../store/actions/downloads";
+import { mvlAttachmentDownloadFromIdSelector } from "../../../../store/reducers/downloads";
+import MVL_ROUTES from "../../../../navigation/routes";
+import { isIos } from "../../../../../../utils/platform";
+import { ContentTypeValues } from "../../../../../../types/contentType";
+import { useIOBottomSheetModal } from "../../../../../../utils/hooks/bottomSheet";
+import ROUTES from "../../../../../../navigation/routes";
+import { mvlPreferencesSelector } from "../../../../store/reducers/preferences";
+
+const BOTTOM_SHEET_HEIGHT = 375;
+
+type BottomSheetProps = Readonly<{
+  /**
+   * Called on right-button press with the user's selected preferences.
+   */
+  onConfirm: (dontAskAgain: boolean) => void;
+
+  /**
+   * The user canceled the action via the UI.
+   */
+  onCancel: () => void;
+}>;
+
+const useDownloadAttachmentConfirmationBottomSheet = ({
+  onConfirm,
+  onCancel
+}: BottomSheetProps) => {
+  const [dontAskAgain, setDontAskAgain] = useState<boolean>(false);
+
+  return useIOBottomSheetModal(
+    <View>
+      <View spacer={true} />
+      <Body>
+        {i18n.t("features.mvl.details.attachments.bottomSheet.warning.body")}
+      </Body>
+      <View spacer={true} />
+      <View style={IOStyles.row}>
+        <RawCheckBox
+          checked={dontAskAgain}
+          onPress={() => setDontAskAgain(!dontAskAgain)}
+        />
+        <Body
+          style={{ paddingLeft: 8 }}
+          onPress={() => setDontAskAgain(!dontAskAgain)}
+        >
+          {i18n.t("features.mvl.details.attachments.bottomSheet.checkBox")}
+        </Body>
+      </View>
+    </View>,
+    i18n.t("features.mvl.details.attachments.bottomSheet.warning.title"),
+    BOTTOM_SHEET_HEIGHT,
+    <FooterWithButtons
+      type={"TwoButtonsInlineHalf"}
+      leftButton={{
+        ...cancelButtonProps(onCancel),
+        onPressWithGestureHandler: true
+      }}
+      rightButton={{
+        ...confirmButtonProps(() => {
+          onConfirm(dontAskAgain);
+        }, i18n.t("global.buttons.continue")),
+        onPressWithGestureHandler: true
+      }}
+    />
+  );
+};
+
+export const useMvlAttachmentDownload = (attachment: MvlAttachment) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatch = useIODispatch();
+  const { navigate } = useNavigation();
```

Comment:
```suggestion
  const { navigate } = useNavigation<IOStackNavigationProp<AppParamsList>>();
```
A little suggestion in order to add some typesafeness to the navigation path (resilient to any future changes).

Reply:
7fb0915e